magento/magento2
# Enable Magento 2 to connect MySQL through SSL.

Enable Magento 2 to connect MySQL through SSL.

### Description
We have made some modification to enable Magento 2 connect MySQL through SSL.

During the installation, there was no option to force MySQL connection through SSL. With this PR, the following options will be enabled for you to use SSL.
```
bin/magento setup:install \
	--db-host=<MYSQL_HOST> \
	--db-name=<MYSQL_DATABASE> \
	--db-user=<MYSQL_USERNAME> \
	--db-password=<MYSQL_PASSWORD> \
	--db-ssl-key=<MYSQL_CLIENT_KEY> \
	--db-ssl-cert=<MYSQL_CLIENT_CERT> \
	--db-ssl-ca=<MYSQL_SERVER_CERT>
```
### Fixed Issues (if relevant)
1. magento/magento2#13561: Magento 2 and SSL connection to MySQL

### Manual testing scenarios
1. Install a MySQL server which accepts connections through SSL.
2. Copy these files into a folder which is accessible by Magento: `client-key.pem`, `client-cert.pem`, `ca.pem`.
3. Install a fresh Magento with the following command-line command:
```
bin/magento setup:install \
	--db-host=<MYSQL_HOST> \
	--db-name=<MYSQL_DATABASE> \
	--db-user=<MYSQL_USERNAME> \
	--db-password=<MYSQL_PASSWORD> \
	--db-ssl-key=<PATH/TO/CLIENT-KEY.pem> \
	--db-ssl-cert=<PATH/TO/CLIENT-CERT.pem> \
	--db-ssl-ca=<PATH/TO/CA.pem>
```
4. Observe that `driver_options` is added into `env.php` file.

### Contribution checklist
